gunrock/util/cuda_properties.cuh

```diff
@@ -14,22 +14,27 @@
 
 #pragma once
 
-namespace gunrock {
-namespace util {
-
 /*****************************************************************
  * Macros for guiding compilation paths
  *****************************************************************/
 
 /**
  * CUDA architecture of the current compilation path
+ *
+ * When compiling nvcc sets the __CUDA_ARCH__ environment variable
+ * to the current compute capability being compiled for.
+ * We set GR_CUDA_ARCH to this value if it's available and 300
+ * otherwise (in host-side code)
  */
-#ifndef __CUDA_ARCH__
-//#define __GR_CUDA_ARCH__ 0                      // Host path
+#ifdef __CUDA_ARCH__ // Device-side compilation
+#define GR_CUDA_ARCH __CUDA_ARCH__
 #else
-#define __GR_CUDA_ARCH__ __CUDA_ARCH__  // Device path
+#define GR_CUDA_ARCH 300 // Host-side compilation
```

Comment:
Why `300` and is it necessary to have this at all for host-side compilation?

Reply:
It's only necessary to get the templates in KernelPolicy structs to compile during the host-side pass. Any number will work, but it just has to exist so that math inside the templates can work. I chose 300 because that's what was used host-side before. 

I agree that it's a hack, so I'd like to either:
- Move the occupancy calculations out of templates and use the occupancy APIs that CUDA provides
- Not compile the structs host side (and so don't do the math at all) with some #ifdefing 

I thought for now it'll get the job done though.

Reply:
Ok, that's a fair explanation. Let's make an issue about this or document this in an existing issue so we remember to get back to it when we switch to use CUDA's occupancy calculator.